### ISSUE TITLE
Grey/White bar on right side of sketches in Processing 4.3 on Manjaro w/ KDE Plasma 6, Wayland

### ISSUE THREAD

**Issue by peterambos** [processing/processing4#862]
_This is my first time writing an issue on a software project, so if I'm doing anything wrong, please let  me know. I've searched for this issue on Google, here and the forums and found nothing so far, and read the guidelines for submitting issues and the troubleshooting page, but if I'm still doing something wrong, please tell me._

## Description
When running a sketch in Processing 4.3, there always is a grey/white bar on the right side of the output, despite it not being coded to be there. The bug persists after uninstalling and reinstalling.

## Expected Behavior
The sketches running without a unwanted bar appearing on the right side of the output.

## Current Behavior
See Description.
Example Screenshot:
![example-screenshot](https://github.com/user-attachments/assets/c3c9f718-c94b-4dec-8122-fbabd072bbb6)

## Steps to Reproduce

1. Open Processing 4.3.
2. This is the smallest amount of code I found to be able to reproduce the bug:
`void setup() {
 size(400, 400);
 background(0);
}`
3. Run the Sketch, which produces the output given in the screenshot given above.

## Your Environment
* Processing version: 4.3 (64 Bit)
* Operating System and OS version: Manjaro Linux 6.10.13 w/ KDE Plasma 6.1.5 on Wayland
* Other information: I've attached a txt with my inxi output for further information.
[sysinfo_inxi.txt](https://github.com/user-attachments/files/17936086/sysinfo_inxi.txt)

I'm happy to provide any further information if needed.

**Comment by Stefterv** [comment]
Hi, quick question, do you use a scaled UI mode on your display, e.g. 125% on 2560x1440

**Comment by peterambos** [comment]
Jup, I use a scale of 120 % on a 2560x1600 display.

**Comment by Stefterv** [comment]
Okay, could you test for me turning it to 100% scale for a second, I suspect the issue is with having non-integer scaling

**Comment by peterambos** [comment]
Sure! The bar is unfortunately still there after turning the scale back to 100%. Even tried logging out and back in to start a new session after changing the scaling, but the bar is still there.

**Comment by Stefterv** [comment]
Okay, thank you so much for trying, will need to be looked into deeper with a vm of your OS then

**Comment by peterambos** [comment]
Got it, if you need any more information, just let me know.

**Comment by peterambos** [comment]
Since there was an update, just wanted to report this issue still persists in Processing 4.3.2 on the 2025-01-18 testing branch update of Manjaro with Plasma 6.2.5.

Current inxi: 
[inxi.txt](https://github.com/user-attachments/files/18469804/inxi.txt)

**Comment by Stefterv** [comment]
Thanks for the ping, I just set up a vm and I'm seeing the same thing:

<img width="1347" alt="Image" src="https://github.com/user-attachments/assets/db730b89-536a-4ef1-b04e-9eaf85c40a58" />

**Comment by Stefterv** [comment]
Switching to P2D solves the issue
<img width="1392" alt="Image" src="https://github.com/user-attachments/assets/a0dd533e-cb3e-4a21-9a32-416bc16b8601" />

**Comment by Stefterv** [comment]
Hi @PeteAmbos I've created a pull request with a potential fix, could you please try the test build created [here](https://github.com/processing/processing4/actions/runs/12856973395l) and let me know if that solves your issue

**Comment by peterambos** [comment]
I can confirm using P2D fixes it on the regular build. I've downloaded and run the test build and unfortunately, it doesn't fix the issue. It still only works when explicitly using P2D. This is the first time I've run a test build though so to make sure I did it correctly, these are the steps I followed:

1.  Downloaded the [processing-21f43731605e0c78373cd9036a1538f550707e7clinux-x64](https://github.com/processing/processing4/actions/runs/12856973395/artifacts/2453434141) zip archive from the page you linked me to.
2. Extracted the tgz file from that zip archive.
3. Followed the steps on the [Getting Started page on processing.org](https://processing.org/tutorials/gettingstarted) as I would for any regular processing install

**Comment by Stefterv** [comment]
Added a different solution, seems like that something on Manjaro is resizing the window even though it is not resizable, causing the lines to appear, this solution sizes the window back down to the sketch size. More robust than the first solution

**Comment by Stefterv** [comment]
@PeteAmbos Thanks so much for testing! I suspect you are doing it correctly, just check if you are running the correct new version by looking at the title bar, it should say `Processing 21f43... ` next to the file name

I've also tested a different approach here: https://github.com/processing/processing4/actions/runs/12857195890 

Oh and please post a screenshot again when testing, there's a lot I can glance from it

**Comment by peterambos** [comment]
@Stefterv Got it! So, I have bad news. On the new build, this is how the IDE looks when running `./processing`:

![Image](https://github.com/user-attachments/assets/79584825-b39b-4e64-bb40-8423282f0a77)

I did manage to copy the error message from the IDE output (in a separate text file due to the length).

[error_msg_ide.txt](https://github.com/user-attachments/files/18470501/error_msg_ide.txt)

I also tried copying `void setup() {  size(400, 400);  background(0); }` into the one line of the editor that is not white and tried to run it using Ctrl + R to see if it would still run a sketch, but no dice, unfortunately.

Also, unlike the other times starting Processing for the first time, no icon appeared on the Desktop. It also doesn't say `Processing 68c8...` in the title bar, but I deleted every other Processing install/folder from my system to make sure it's the only one that could be running right now.

**Comment by Stefterv** [comment]
Fascinating, yeah that is still 4.3.2 for sure, probably erroring out because it's source files have disappeared 😅

I suspect closing Processing didn't fully close it and somehow it is still running in the background somewhere. Maybe restart too? 

I started working on a tool this weekend to make the experience of testing Processing builds easier 

Thanks again

**Comment by peterambos** [comment]
Restarting did the trick, and the white bar is now gone as well!

![Image](https://github.com/user-attachments/assets/b4fdd1ae-ed8e-4499-aa2b-50e374d42e0c)

**Comment by Stefterv** [comment]
Yay! 


Sorry I’m keeping this open until the PR is merged

**Comment by peterambos** [comment]
Oh, sorry, I didn't know it was supposed to remain open until then, I'll remember that for the future.

**Comment by Stefterv** [comment]
Yep, I've linked the PR in such a way that merging it will close the issue automatically 😉

**Comment by peterambos** [comment]
@Stefterv by the way, I could also test the test build on Windows 11, if that would help make sure it is cross-compatible.

**Comment by Stefterv** [comment]
Cool yeah please do!

**Comment by peterambos** [comment]
I apologize for the delay, but I've managed to test it now, and it seems to work without any problems on Windows 11.

![Image](https://github.com/user-attachments/assets/3aba112e-a474-4793-a68a-4105d1d146b4)

Here's my DxDiag for further info about my system:

[DxDiag.txt](https://github.com/user-attachments/files/18482559/DxDiag.txt)
